doc/implicit_conversion.rdoc

```diff
@@ -0,0 +1,187 @@
+== Implicit Conversion
+
+Some Ruby methods accept one or more objects
+that can be either:
+* <i>Of a given class</i>, and so accepted as is.
+* <i>Implicitly convertible to that class</i>, in which case
+  the called method converts the object.
+
+For each of the relevant classes, the conversion is done by calling
+a specific conversion method:
+
+* Array: +to_ary+
+* Hash: +to_hash+
+* Integer: +to_int+
+* String: +to_str+
+
+=== Array-Convertible Objects
```

Comment:
"Array-Convertible Object" is not an actual term in Ruby. If we introduce it, we should be confident it's the right one.

An alternative would be "Array Coercion" or "Array Type Conversion".

I also understand you are trying to find terminology that is understandable at face value. So, again, I'll leave this to you. However, introducing new terminology also carries some extra weight.